micro-rdk/src/common/ota.rs

```diff
@@ -172,6 +170,17 @@ pub(crate) struct OtaService<S: OtaMetadataStorage> {
 }
 
 impl<S: OtaMetadataStorage> OtaService<S> {
+    pub(crate) async fn stored_metadata(&self) -> OtaMetadata {
```

Comment:
Is this function really `async`?

Reply:
It isn't, but shouldn't it be? The underlying calls are essentially I/O, getting data from flash storage. But this is maybe is more of a "should our storage APIs be async?" question.

I'll remove it here though.